File: Apps/iOS/Domain/Entities/Scores/FoxChase.swift
extension Score {
    static let foxChase = Score(
        name: "Fox chase",
        key: .g,
        position: .second,
        beatsPerMinute: 132,
        events: horn + gallop + whooping + gallop + baying
    )

    private static let horn: [ScoreEvent] = [
        .blow([.one, .two, .three], 1.5),
        .blow([.one, .two, .three], 0.5),
        .blow([.one, .two, .three], 2, vibrato: 0.7),
        .rest(beats: 1)
    ]

    private static let gallop: [ScoreEvent] = [
        .draw([.one, .two], 0.333), .draw([.one, .two], 0.333), .blow([.one, .two], 0.334),
        .draw([.one, .two], 0.333), .draw([.one, .two], 0.333), .blow([.one, .two], 0.334),
        .draw([.one, .two], 0.333), .draw([.one, .two], 0.333), .blow([.one, .two], 0.334),
        .draw([.one, .two], 0.333), .draw([.one, .two], 0.333), .blow([.one, .two], 0.334)
    ]

    private static let whooping: [ScoreEvent] = [
        .draw(.four, 1, shakenWith: .five),
        .draw(.five, 1, shakenWith: .six),
        .draw(.two, 1, bentBy: 2, releasingTo: 0),
        .draw(.three, 1, bentBy: 3, releasingTo: 0)
    ]

    private static let baying: [ScoreEvent] = [
        .draw(.four, 1.5, shakenWith: .five),
        .draw(.three, 1, bentBy: 3, releasingTo: 1),
        .draw(.two, 1.5, bentBy: 2, vibrato: 0.9),
        .rest(beats: 2)
    ]
}

File: Web/src/domain/scores/tunes/foxChase.ts
import type { Score, ScoreEvent } from '../score.js'
import { blow, draw, rest } from '../scoreWriting.js'

const horn: ScoreEvent[] = [
    blow([1, 2, 3], 1.5),
    blow([1, 2, 3], 0.5),
    blow([1, 2, 3], 2, { vibrato: 0.7 }),
    rest(1)
]

const gallop: ScoreEvent[] = [
    draw([1, 2], 0.333), draw([1, 2], 0.333), blow([1, 2], 0.334),
    draw([1, 2], 0.333), draw([1, 2], 0.333), blow([1, 2], 0.334),
    draw([1, 2], 0.333), draw([1, 2], 0.333), blow([1, 2], 0.334),
    draw([1, 2], 0.333), draw([1, 2], 0.333), blow([1, 2], 0.334)
]

const whooping: ScoreEvent[] = [
    draw(4, 1, { shakenWith: 5 }),
    draw(5, 1, { shakenWith: 6 }),
    draw(2, 1, { bentBy: 2, releasingTo: 0 }),
    draw(3, 1, { bentBy: 3, releasingTo: 0 })
]

const baying: ScoreEvent[] = [
    draw(4, 1.5, { shakenWith: 5 }),
    draw(3, 1, { bentBy: 3, releasingTo: 1 }),
    draw(2, 1.5, { bentBy: 2, vibrato: 0.9 }),
    rest(2)
]

export const foxChase: Score = {
    name: 'Fox chase',
    key: 'G',
    position: 'second',
    beatsPerMinute: 132,
    events: [...horn, ...gallop, ...whooping, ...gallop, ...baying]
}
